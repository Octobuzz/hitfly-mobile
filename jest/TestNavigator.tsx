// https://github.com/react-navigation/rfcs/issues/75
import React, { Children } from 'react'
import {
  createAppContainer,
  createSwitchNavigator,
  NavigationParams,
} from 'react-navigation'

const TestNavigator = ({
  children,
  params,
}: {
  children: any
  params?: NavigationParams
}) => {
  const Navigator = createAppContainer(
    createSwitchNavigator({
      TestScreen: { screen: () => Children.only(children), params },
    }),
  )
  return <Navigator />
}

export default TestNavigator
