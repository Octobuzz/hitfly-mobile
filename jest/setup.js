import 'jest-styled-components'
import mockAsyncStorage from '@react-native-community/async-storage/jest/async-storage-mock'

jest.mock('react-native-vector-icons/Foundation', () => 'Icon')
jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'Icon')
jest.mock('react-native-vector-icons/Feather', () => 'Icon')
jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon')
jest.mock('react-native-vector-icons/SimpleLineIcons', () => 'Icon')

jest.mock('@react-native-community/async-storage', () => mockAsyncStorage)

jest.mock('react-native-track-player', () => {
  return {
    addEventListener: jest.fn(),
    registerEventHandler: jest.fn(),
    registerPlaybackService: jest.fn(),
    setupPlayer: jest.fn(),
    destroy: jest.fn(),
    updateOptions: jest.fn(),
    add: jest.fn(),
    remove: jest.fn(),
    skip: jest.fn(),
    skipToNext: jest.fn(),
    skipToPrevious: jest.fn(),
    removeUpcomingTracks: jest.fn(),
    // playback commands
    reset: jest.fn(),
    play: jest.fn(),
    pause: jest.fn(),
    stop: jest.fn(),
    seekTo: jest.fn(),
    setVolume: jest.fn(),
    setRate: jest.fn(),
    // player getters
    getQueue: jest.fn(),
    getTrack: jest.fn(),
    getCurrentTrack: jest.fn(),
    getVolume: jest.fn(),
    getDuration: jest.fn(),
    getPosition: jest.fn(),
    getBufferedPosition: jest.fn(),
    getState: jest.fn(),
    getRate: jest.fn(),
  }
})

jest.mock('react-native-gesture-handler', () => {
  const View = require('react-native/Libraries/Components/View/View')
  return {
    Swipeable: View,
    DrawerLayout: View,
    State: {},
    ScrollView: View,
    Slider: View,
    Switch: View,
    TextInput: View,
    ToolbarAndroid: View,
    ViewPagerAndroid: View,
    DrawerLayoutAndroid: View,
    WebView: View,
    NativeViewGestureHandler: View,
    TapGestureHandler: View,
    FlingGestureHandler: View,
    ForceTouchGestureHandler: View,
    LongPressGestureHandler: View,
    PanGestureHandler: View,
    PinchGestureHandler: View,
    RotationGestureHandler: View,
    /* Buttons */
    RawButton: View,
    BaseButton: View,
    RectButton: View,
    BorderlessButton: View,
    /* Other */
    FlatList: View,
    gestureHandlerRootHOC: jest.fn(),
    Directions: {},
  }
})
