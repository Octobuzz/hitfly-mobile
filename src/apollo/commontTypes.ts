export type CollectionsType = 'recommended' | 'musicFan'
