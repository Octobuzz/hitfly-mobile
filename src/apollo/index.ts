import * as _storage from './storage'
export const storage = _storage

export * from './queries'
export * from './schemas'
export * from './mutations'
export * from './commonTypes'

export { default as setupClient } from './setupClient'
