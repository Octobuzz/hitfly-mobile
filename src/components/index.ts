export * from './views'
export * from './fields'
export * from './buttons'

export { default as TextBase } from './TextBase'
export { default as GenreItem } from './GenreItem'
export { default as TextWithLines } from './TextWithLines'
