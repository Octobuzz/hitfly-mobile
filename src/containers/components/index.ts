export * from './SocialAuth'

export { default as SocialAuth } from './SocialAuth'
export { default as HeaderRightButtons } from './HeaderRightButtons'
