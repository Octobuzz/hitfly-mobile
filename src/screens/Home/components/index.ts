export { default as StarsSection } from './StarsSection'
export { default as TracksSection } from './TracksSection'
export { default as SectionHeader } from './SectionHeader'
export { default as SectionWrapper } from './SectionWrapper'
export { default as PlaylistHeader } from './PlaylistHeader'
export { default as CollectionSection } from './CollectionSection'
