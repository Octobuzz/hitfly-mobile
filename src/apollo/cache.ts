import { InMemoryCache } from 'apollo-cache-inmemory'
import { persistCache } from 'apollo-cache-persist'
import { storageInstance } from 'src/utils'
import gql from 'graphql-tag'

export const defaults = {
  headerSettings: {
    mode: 'dark',
    state: 'main',
    __typename: 'HeaderSettings',
  },
  isPlaying: false,
  activeTrack: null,
  playlist: [],
}

export const typeDefs = gql`
  type Track {
    id: Int!
    title: String
    singer: String
    date: String
    songText: String
    fileUrl: String
    musicWave: [Int]
    length: Int
  }
  type HeaderSettings {
    style: String!
    state: String!
  }
`

export default async (): Promise<InMemoryCache> => {
  const cache = new InMemoryCache({
    freezeResults: true,
  })

  await persistCache({
    cache,
    debug: __DEV__,
    // @ts-ignore
    storage: storageInstance,
  })

  return cache
}
