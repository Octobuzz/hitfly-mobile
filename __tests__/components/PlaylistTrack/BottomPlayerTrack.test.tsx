import React from 'react'
import { render, fireEvent } from '../../../jest/test-utils'
import { BottomPlayerTrack } from 'src/components'
import { Track } from 'src/apollo'

describe('BottomPlayerTrack', () => {
  const track = {
    id: 1,
    cover: [
      {
        imageUrl: 'some url',
      },
    ],
    title: 'Some track title',
    group: {
      title: 'Some group title',
    },
  } as Track

  it('renders correctly', () => {
    const { asJSON } = render(<BottomPlayerTrack track={track} />)
    expect(asJSON()).toMatchSnapshot()
  })

  it('renders correctly in playing state', () => {
    const { asJSON } = render(<BottomPlayerTrack isPlaying track={track} />)
    expect(asJSON()).toMatchSnapshot()
  })

  it('can be pressed', () => {
    const onPress = jest.fn()
    const { getByTestId } = render(
      <BottomPlayerTrack onPress={onPress} track={track} />,
    )
    fireEvent.press(getByTestId('wrapper'))
    expect(onPress).toBeCalled()
  })

  it('can be pressed in playing state', () => {
    const onPress = jest.fn()
    const { getByTestId } = render(
      <BottomPlayerTrack isPlaying onPress={onPress} track={track} />,
    )
    fireEvent.press(getByTestId('wrapper'))
    expect(onPress).toBeCalled()
  })

  it('allows to press control', () => {
    const onPressControl = jest.fn()
    const { getByTestId } = render(
      <BottomPlayerTrack onPressControl={onPressControl} track={track} />,
    )
    fireEvent.press(getByTestId('control'))
    expect(onPressControl).toBeCalled()
  })

  it('allows to press control in playing state', () => {
    const onPressControl = jest.fn()
    const { getByTestId } = render(
      <BottomPlayerTrack
        isPlaying
        onPressControl={onPressControl}
        track={track}
      />,
    )
    fireEvent.press(getByTestId('control'))
    expect(onPressControl).toBeCalled()
  })
})
