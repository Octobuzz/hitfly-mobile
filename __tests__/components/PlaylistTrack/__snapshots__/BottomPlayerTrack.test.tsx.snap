// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomPlayerTrack renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.8}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
  testID="wrapper"
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "borderRadius": 4,
            "height": 24,
            "width": 24,
          },
        ],
      ]
    }
  >
    <FastImageView
      resizeMode="cover"
      source={
        Object {
          "uri": "some url",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginHorizontal": 16,
        },
      ]
    }
  >
    <Text
      numberOfLines={2}
      style={
        Array [
          Object {
            "color": "#313131",
            "fontFamily": "GothamPro-Bold",
            "fontSize": 12,
            "lineHeight": 18,
          },
        ]
      }
    >
      Some track title
    </Text>
    <Text
      numberOfLines={2}
      style={
        Array [
          Object {
            "color": "#606060",
            "fontFamily": "GothamPro",
            "fontSize": 10,
            "lineHeight": 12,
          },
        ]
      }
    />
  </View>
  <TouchableOpacity
    activeOpacity={0.2}
    style={
      Array [
        Object {
          "alignItems": "center",
          "width": 34,
        },
      ]
    }
    testID="control"
  >
    <Icon
      color="#313131"
      name="play"
      size={36}
      style={
        Array [
          Object {},
        ]
      }
    />
  </TouchableOpacity>
</TouchableOpacity>
`;

exports[`BottomPlayerTrack renders correctly in playing state 1`] = `
<TouchableOpacity
  activeOpacity={0.8}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
  testID="wrapper"
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "borderRadius": 4,
            "height": 24,
            "width": 24,
          },
        ],
      ]
    }
  >
    <FastImageView
      resizeMode="cover"
      source={
        Object {
          "uri": "some url",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginHorizontal": 16,
        },
      ]
    }
  >
    <Text
      numberOfLines={2}
      style={
        Array [
          Object {
            "color": "#313131",
            "fontFamily": "GothamPro-Bold",
            "fontSize": 12,
            "lineHeight": 18,
          },
        ]
      }
    >
      Some track title
    </Text>
    <Text
      numberOfLines={2}
      style={
        Array [
          Object {
            "color": "#606060",
            "fontFamily": "GothamPro",
            "fontSize": 10,
            "lineHeight": 12,
          },
        ]
      }
    />
  </View>
  <TouchableOpacity
    activeOpacity={0.2}
    style={
      Array [
        Object {
          "alignItems": "center",
          "width": 34,
        },
      ]
    }
    testID="control"
  >
    <Icon
      color="#313131"
      name="pause"
      size={36}
      style={
        Array [
          Object {},
        ]
      }
    />
  </TouchableOpacity>
</TouchableOpacity>
`;
